src/Validator.php

```diff
@@ -405,6 +405,23 @@ protected function validateRef(&$document_data, &$data, array $data_pointer, arr
             unset($pointer);
         }
 
+        // Check if we can find the fragment through the loader
```

Comment:
Here `$ref` is not normalized
`../file.json` == `../a/../file.json` == `../b/../file.json` ...
The loader should receive the normalized version of the uri.